surefire-provider/src/main/java/org/arquillian/smart/testing/surefire/provider/StrategyComparator.java

```diff
@@ -0,0 +1,31 @@
+package org.arquillian.smart.testing.surefire.provider;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.arquillian.smart.testing.TestSelection;
+
+class StrategyComparator implements Comparator<TestSelection> {
+    private final Map<String, Integer> priorities;
+    private final List<String> strategies;
+
+    StrategyComparator(List<String> strategies) {
+        this.strategies = strategies;
+        this.priorities = priority();
+    }
+
+    private Map<String, Integer> priority() {
+        return strategies.stream()
+            .collect(Collectors.toMap(Function.identity(), s -> (strategies.size() - strategies.indexOf(s)) * 100));
```

Comment:
Consider following case: we have `[a,b,c,d,e]` strategies. Having this coding tests falling into `[b,e]` (`(5-1) * 100 + (5-4) * 100 = 400 + 100 = 500`) are equal to those which are in `[a]` (`500`) category which is not necessarily what we want.

I would rather think of having some sort of "bitset" representation here, where `i` being a position on the list (but starting from 1) would be treated as `10^(size-i)`. Then having `[a,b,c,d,e]` a test falling into `[b,e]` would be equal to 01001 and the one failling into `[a]` would be 10000, so comparing it would result in `[a]` being executed before `[b,e]` as desired.

Or am I missing something here?